tests/test-shape.lua

```diff
@@ -0,0 +1,54 @@
+-- Test if shape crash when called
+-- Also generate some SVG to be used by the documentation
+-- it also "prove" that the code examples are all working
+local cairo = require( "lgi"         ).cairo
+local shape = require( "gears.shape" )
+local filepath, svgpath = ...
+
+print("\n\n\n",filepath, svgpath)
+
+local function get_surface(p)
+    local img = cairo.SvgSurface.create(p, 288, 76)
+    return cairo.Context(img)
+end
+
+local function show(cr, skip_fill)
+    cr:set_source_rgba(0.380392156863,0.505882352941,1,1)
+
+    if skip_fill then
+        cr:stroke()
+    else
+        cr:stroke_preserve()
+        cr:set_source_rgba(0.380392156863,0.505882352941,1,0.5)
+        cr:fill()
```

Comment:
Doesn't this fill partly draw above the stroke, which should be visible thanks to transparency? I'd do it the other way around, first fill, then stroke (which doesn't have any transparency) to avoid this. Or is that on purpose?